README.md

```diff
@@ -1,251 +1,395 @@
+city_state : Anywhere, USA
+zip_code : 00409
+```
 
+This brings us to our second problem: what if we want add a new piece of information, like Jenny's last name? We have to keep the order of not only both the `keys` array and the `jenny` array synchronized, but we have to keep the order of every other contact array synchronized as well:
 
 ```objc
-NSString *phoneNumberForFriendNamedJoe = friendPhonesDictionary["Joe"];
+NSArray *keys = @[ @"first_name",
+                   @"last_name",
+                   @"relationship",
+                   @"phone_number",
+                   @"email_address",
+                   @"physical_address",
+                   @"city_state",
+                   @"zip_code"         ];
+
+
+NSArray *jenny = @[ @"Jenny",
+                    @"Curran",
+                    @"Friend",
+                    @"(555) 867-5309",
+                    @"jenny@email.com",
+                    @"123 Street Name",
+                    @"Anywhere, USA",
+                    @"00409"           ];
+    
+
+NSArray *unclebob = @[ @"Bob",
+                       @"",       // we must insert an empty string as a placeholder
+                       @"Uncle",
+                       @"(555) 876-1234",
+                       @"unclebob@email.com",
+                       @"234 Street Name",
+                       @"Somewhere, USA",
+                       @"00201"           ];
 ```
 
-A contact list consists of a series of names, and a series of phone numbers. It is standard to look up a phone number value by its "key", the contact's name.
+This is a lot of work, and if we miss adding a field to just one contact's array we could cause a crash due to a short index; or if we add it in the wrong place, then the information won't line up correctly with their descriptions in the `keys` array. With these small groups of information, it doesn't seem like an unmanageable task, but when dealing with thousands of groups, each with hundreds of information fields, these manual adjustments become untenable.
 
+Enter key-value pairing.
 
-## Creating an `NSDictionary`
+## Creating An `NSDictionary`
 
-Let's move ahead with the example of a contact list with names and phone numbers. A dictionary creates pairs for these names and numbers such that we can search for a phone number, given a contact's name. Here is a sample of what creating that `NSDictionary` would look like, using the [literal syntax](http://cocoaheads.tumblr.com/post/17757846453/objective-c-literals-for-nsdictionary-nsarray) for dictionaries. (We'll store our phone numbers as strings initially as well.)
+Dictionaries are a collection type common to most programming languages. They store an unordered group of key-value pairs. A key-value pair is a binding of two objects: the descriptive accessor of a piece of information (the "key"), and the information itself (the "value"). The base dictionary type in Objective-C is `NSDictionary` and has its own literal syntax that looks like this:
```

Comment:
Another way to put it is to think of it like a Dictionary. You want the definition of "salad" you look up the key of "salad" and get the value of "vegetable based thing"